File: poserp/lib/modules/khaata/services/khaata_service.dart
import '../../../../core/api/api_client.dart';
import '../models/khaata_party.dart';

class KhaataService {
  final ApiClient _apiClient;

  KhaataService(this._apiClient);

  Future<List<KhaataParty>> getBalances() async {
    final response = await _apiClient.get('/khaata/balances');
    final body = response.data as Map<String, dynamic>;
    final list = body['data'] as List? ?? [];
    return list
        .map((e) => KhaataParty.fromJson(e as Map<String, dynamic>))
        .toList();
  }

  Future<Map<String, dynamic>> addTransaction(
    String partyId, {
    required double amount,
    required String type,
    required String partyType,
    String? notes,
  }) async {
    final response = await _apiClient.post(
      '/khaata/$partyId/transaction',
      data: {
        'amount': amount,
        'type': type,
        'partyType': partyType,
        'notes': notes ?? '',
      },
    );
    return response.data as Map<String, dynamic>;
  }

  Future<void> logReminder(String partyId, String message) async {
    await _apiClient.post(
      '/khaata/$partyId/remind',
      data: {'message': message},
    );
  }
}

File: src/services/khaataService.ts
import api from "@/services/api";

export interface KhaataParty {
  _id: string;
  name: string;
  phone: string;
  partyType: 'customer' | 'supplier';
  currentBalance: number;
  creditDays: number;
  isAutoReminderEnabled: boolean;
  lastReminderSentAt: string | null;
}

export interface KhaataTransaction {
  _id: string;
  partyId: string;
  partyType: string;
  type: string;
  debitAmount: number;
  creditAmount: number;
  balanceAfter: number;
  date: string;
  status: string;
}

export const khaataService = {
  getBalances: async () => {
    const response = await api.get<{ success: boolean; data: KhaataParty[] }>("/khaata/balances");
    return response.data.data;
  },

  addTransaction: async (partyId: string, payload: { amount: number; type: 'payment_in' | 'payment_out'; notes?: string; partyType: 'customer' | 'supplier' }) => {
    const response = await api.post<{ success: boolean; data: KhaataTransaction }>(`/khaata/${partyId}/transaction`, payload);
    return response.data.data;
  },

  logReminder: async (partyId: string, message: string) => {
    const response = await api.post(`/khaata/${partyId}/remind`, { message });
    return response.data;
  },
};

